x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -85,6 +73,29 @@ export function isMappableJob(job: CombinedJob, detectorIndex: number): boolean
   return isMappable;
 }
 
+export function hasValidComposite(buckets: { [key: string]: any }) {
+  if (
+    isPopulatedObject(buckets, ['composite']) &&
+    isPopulatedObject(buckets.composite, ['sources'])
+  ) {
+    const sources = buckets.composite.sources as Array<Record<string, object>>;
```

Comment:
I think manual typecasting might be dangerous and we should use a type guard instead

Reply:
You're right that we don't need manual typecasting here. Fixed here [`2a9d1fc` (#96348)](https://github.com/elastic/kibana/pull/96348/commits/2a9d1fc773ad0fe52f49b62d1fb042d803b56013)